### PR TITLE
Internal: Conversion Center - Contact Buttons Widget

### PR DESCRIPTION
**Pull request by nicoladj77**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Added First variation of Contatc Buttons widget

## Description
An explanation of what is done in this PR

* Added First variation of Contatc Buttons widget

## Test instructions
This PR can be tested by following these steps:

* Add Widget to editor and test

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/frontend/frontend.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Breakpoints from 'elementor-utils/breakpoints';
 import Events from 'elementor-utils/events';
 import Shapes from 'elementor/modules/shapes/assets/js/frontend/frontend';
 import Controls from './utils/controls';
+import ConversionCenter from 'elementor/modules/conversion-center/assets/js/frontend';
```
Comment (louiswol94) [inline assets/dev/js/frontend/frontend.js:17]: Load this behind the Experiment in this file:

https://github.com/elementor/elementor/blob/main/assets/dev/js/frontend/elements-handlers-manager.js#L33-L35

File: modules/conversion-center/assets/js/handlers/contact-buttons.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+			},
+		};
+	}
+
+	getDefaultElements() {
+		const selectors = this.getSettings( 'selectors' );
+		return {
+			$content: this.$element.find( selectors.content ),
+			$chatButton: this.$element.find( selectors.chatButton ),
+			$closeButton: this.$element.find( selectors.closeButton ),
```
Comment (louiswol94) [inline modules/conversion-center/assets/js/handlers/contact-buttons.js:17]: We try to use native JS these days:

`content: this.$element[ 0 ].querySelector( selectors. content ),`

And:

`this.elements.content.classList.toggle( 'is-visible' );`

And:

`this.elements.closeButton.addEventListener( 'click', this. onChatButtonClick.bind( this ) );`

You can decide if you want to re-write this (while it's small), or just keep it in mind for future handlers.

(We want to get rid of jQuery someday, or at least defer the loading of jQuery, so it helps if as little as possible widgets depend on it).

I will leave it up to you to decide :)

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+
+	public function get_keywords(): array {
+		return [ 'buttons', 'contact', 'widget' ];
+	}
+
+	public function get_stack( $with_common_controls = true ): array {
+		return parent::get_stack( false );
+	}
+
+
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-contact-button-base.php:33]:
```suggestion
```
Remove double lines

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+		// Styles
+		$this->add_style_chat_button_section();
+
+		$this->add_style_top_bar_section();
+
+		$this->add_advanced_tab();
+	}
+
+	// Controls
+
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-contact-button-base.php:50]:
```suggestion
```
Remove

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+				'return_value' => 'yes',
+				'default' => 'yes',
+			]
+		);
+
+		$this->end_controls_section();
+	}
+
+	// Styles
+
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-contact-button-base.php:302]:
```suggestion
```
Remove

File: modules/conversion-center/classes/providers/social-network-provider.php [hunk 6]
```diff
@@ -52,6 +55,17 @@ public static function get_icon_mapping( string $platform ): string {
...
 
+	public static function get_name_mapping( string $platform ): string {
+		static::init_social_networks_array_if_empty();
+
+		if ( isset( self::$social_networks[ $platform ]['name'] ) ) {
+			return self::$social_networks[ $platform ]['name'];
+		}
+
+		return '';
+
```
Comment (louiswol94) [inline modules/conversion-center/classes/providers/social-network-provider.php:66]:
```suggestion
```
Remove

File: modules/conversion-center/classes/render/contact-buttons-core-render.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Elementor\Modules\ConversionCenter\Classes\Render;
+
+use Elementor\Icons_Manager;
+use Elementor\Modules\ConversionCenter\Classes\Providers\Social_Network_Provider;
+use Elementor\Utils;
```
Comment (louiswol94) [inline modules/conversion-center/classes/render/contact-buttons-core-render.php:7]:
```suggestion
```
I think all of these are unused imports?

File: modules/conversion-center/widgets/contact-buttons.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+namespace Elementor\Modules\ConversionCenter\Widgets;
+
+use Elementor\Controls_Manager;
+use Elementor\Modules\ConversionCenter\Base\Widget_Contact_Button_Base;
+use Elementor\Modules\ConversionCenter\Classes\Providers\Social_Network_Provider;
+use Elementor\Modules\ConversionCenter\Classes\Render\Contact_Buttons_Core_Render;
+use Elementor\Group_Control_Typography;
+use Elementor\Plugin;
+use Elementor\Utils;
+use Elementor\Widget_Base;
```
Comment (louiswol94) [inline modules/conversion-center/widgets/contact-buttons.php:12]:
```suggestion
use Elementor\Modules\ConversionCenter\Base\Widget_Contact_Button_Base;
use Elementor\Modules\ConversionCenter\Classes\Render\Contact_Buttons_Core_Render;
```
Remove unused imports

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/conversion-center/assets/js/handlers/contact-buttons.js (hunk 2 above)
Comment (mserino) [reply modules/conversion-center/assets/js/handlers/contact-buttons.js]: I am absolutely in favour of using Vanilla JS, so I'm removing jQuery now.
I did see jQuery used in all the other handlers and I did the same, but yessss let's remove it!!
Thank you!

File: assets/dev/js/frontend/frontend.js (hunk 1 above)
Comment (mserino) [reply assets/dev/js/frontend/frontend.js]: this is done. Thanks for your help!!

File: modules/conversion-center/assets/js/handlers/contact-buttons.js (hunk 2 above)
Comment (mserino) [reply modules/conversion-center/assets/js/handlers/contact-buttons.js]: Done

File: modules/conversion-center/base/widget-contact-button-base.php (hunk 3 above)
Comment (mserino) [reply modules/conversion-center/base/widget-contact-button-base.php]: done

File: modules/conversion-center/base/widget-contact-button-base.php (hunk 4 above)
Comment (mserino) [reply modules/conversion-center/base/widget-contact-button-base.php]: done

File: modules/conversion-center/base/widget-contact-button-base.php (hunk 5 above)
Comment (mserino) [reply modules/conversion-center/base/widget-contact-button-base.php]: done

File: modules/conversion-center/classes/providers/social-network-provider.php (hunk 6 above)
Comment (mserino) [reply modules/conversion-center/classes/providers/social-network-provider.php]: done

File: modules/conversion-center/classes/render/contact-buttons-core-render.php (hunk 7 above)
Comment (mserino) [reply modules/conversion-center/classes/render/contact-buttons-core-render.php]: removed

File: modules/conversion-center/widgets/contact-buttons.php (hunk 8 above)
Comment (mserino) [reply modules/conversion-center/widgets/contact-buttons.php]: removed

File: modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+			},
+		};
+	}
+
+	getDefaultElements() {
+		const selectors = this.getSettings( 'selectors' );
+		return {
+			$content: this.$element[ 0 ].querySelector( selectors.content ),
+			$chatButton: this.$element[ 0 ].querySelector( selectors.chatButton ),
+			$closeButton: this.$element[ 0 ].querySelector( selectors.closeButton ),
```
Comment (louiswol94) [inline modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js:19]:
```suggestion
			content: this.$element[ 0 ].querySelector( selectors.content ),
			chatButton: this.$element[ 0 ].querySelector( selectors.chatButton ),
			closeButton: this.$element[ 0 ].querySelector( selectors.closeButton ),
```

File: modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+		};
+	}
+
+	bindEvents() {
+		this.elements.$closeButton.addEventListener( 'click', this.onCloseButtonClick.bind( this ) );
+		this.elements.$chatButton.addEventListener( 'click', this.onChatButtonClick.bind( this ) );
```
Comment (louiswol94) [inline modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js:25]:
```suggestion
		this.elements.closeButton.addEventListener( 'click', this.onCloseButtonClick.bind( this ) );
		this.elements.chatButton.addEventListener( 'click', this.onChatButtonClick.bind( this ) );
```

File: modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	onChatButtonClick() {
+		this.elements.$content.classList.toggle( 'is-visible' );
```
Comment (louiswol94) [inline modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js:29]:
```suggestion
		this.elements.content.classList.toggle( 'is-visible' );
```

File: modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	onCloseButtonClick() {
+		this.elements.$content.classList.remove( 'is-visible' );
```
Comment (louiswol94) [inline modules/conversion-center/assets/js/frontend/handlers/contact-buttons.js:33]:
```suggestion
		this.elements.content.classList.remove( 'is-visible' );
```

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**